src/Features/CSharp/Portable/UsePatternCombinators/CSharpUsePatternCombinatorsCodeFixProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Roslyn.Utilities;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternCombinators
+{
+    using static SyntaxFactory;
+    using static AnalyzedPattern;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpUsePatternCombinatorsCodeFixProvider : SyntaxEditorBasedCodeFixProvider
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpUsePatternCombinatorsCodeFixProvider()
+            : base(supportsFixAll: true)
+        {
+        }
+
+        private static SyntaxKind MapToSyntaxKind(BinaryOperatorKind kind) => kind switch
+        {
+            BinaryOperatorKind.LessThan => SyntaxKind.LessThanToken,
+            BinaryOperatorKind.GreaterThan => SyntaxKind.GreaterThanToken,
+            BinaryOperatorKind.LessThanOrEqual => SyntaxKind.LessThanEqualsToken,
+            BinaryOperatorKind.GreaterThanOrEqual => SyntaxKind.GreaterThanEqualsToken,
```

Comment:
i feel like we have this in a few places.  could we move this helper to another directory and share any of this common stuff in a static class?